src/Symfony/Component/Asset/VersionStrategy/RemoteJsonManifestVersionStrategy.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\VersionStrategy;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * Reads the versioned path of an asset from a remote JSON manifest file.
+ *
+ * For example, the manifest file might look like this:
+ *     {
+ *         "main.js": "main.abc123.js",
+ *         "css/styles.css": "css/styles.555abc.css"
+ *     }
+ *
+ * You could then ask for the version of "main.js" or "css/styles.css".
+ */
+class RemoteJsonManifestVersionStrategy implements VersionStrategyInterface
+{
+    private $manifestData;
+    private $httpResponse;
+
+    /**
+     * @param string $manifestUrl Absolute url to the manifest file
+     */
+    public function __construct(string $manifestUrl, HttpClientInterface $httpClient)
+    {
+        $this->httpResponse = $httpClient->request('GET', $manifestUrl);
+    }
+
+    /**
+     * With a manifest, we don't really know or care about what
+     * the version is. Instead, this returns the path to the
+     * versioned file.
+     */
+    public function getVersion(string $path)
+    {
+        return $this->applyVersion($path);
+    }
+
+    public function applyVersion(string $path)
+    {
+        return $this->getManifestPath($path) ?: $path;
+    }
+
+    private function getManifestPath(string $path): ?string
+    {
+        if (null === $this->manifestData) {
+            $this->manifestData = json_decode($this->httpResponse->getContent(), true);
```

Comment:
better use `->toArray()` and catching the DecodingExceptionInterface, don't you think?

Reply:
Is there any way to serve the cache if the response is not a valid JSON ?

Reply:
what does "serve the cache" mean here?

Reply:
I mean, using `stale-if-error` cache directive, if the http request fails (timeout, 500 ...) the cached response should be returned by the http client. But if the clients returns an invalid response with status code 200 (not a JSON response). I wonder if the stale cache could be used. I the current use-case, this is a valuable behavior.